examples/DotNetCore/ConsoleAppWithTableStorageReference/Program.cs

```diff
@@ -0,0 +1,161 @@
+﻿using Azure;
+using Azure.Data.AppConfiguration;
+using Azure.Data.Tables;
+using Azure.Identity;
+using System.Text;
+using System.Text.Json;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.Examples.ConsoleAppWithTableStorageReference
+{
+    // C# record type for items in the table
+    public record Product : ITableEntity
+    {
+        public string RowKey { get; set; } = default!;
+
+        public string PartitionKey { get; set; } = default!;
+
+        public string Name { get; init; } = default!;
+
+        public int Quantity { get; init; }
+
+        public bool OnSale { get; init; }
+
+        public ETag ETag { get; set; } = default!;
+
+        public DateTimeOffset? Timestamp { get; set; } = default!;
+    }
+
+    class Program
+    {
+        static IConfiguration Configuration { get; set; }
+
+        static void Main(string[] args)
+        {
+            Configure();
+
+            string display = string.Empty;
+
+            StringBuilder sb = BuildProductTable();
+
+            sb.AppendLine();
+            sb.AppendLine("Press any key to exit...");
+
+            Console.Clear();
+            Console.Write(sb.ToString());            
+
+            // Finish on key press
+            Console.ReadKey();
+        }
+
+        private static void Configure()
+        {
+            var builder = new ConfigurationBuilder();
+
+            builder.AddJsonFile("appsettings.json")
+                    .AddEnvironmentVariables();
+
+            IConfiguration configuration = builder.Build();
+
+            if (string.IsNullOrEmpty(configuration["ConnectionString"]))
+            {
+                Console.WriteLine("Connection string not found.");
+                Console.WriteLine("Please set the 'ConnectionString' environment variable to a valid Azure App Configuration connection string and re-run this example.");
+                return;
+            }
+
+            // Augment the configuration builder with Azure App Configuration
+            // Pull the connection string from an environment variable
+            builder.AddAzureAppConfiguration(options =>
+            {
+                options.Connect(configuration["ConnectionString"])
+                        .Select("*")
+                        .Map(async (setting) =>
+                        {
+                            if (setting.ContentType.Equals("application/storage.table"))
+                            {
+                                // Example value for key "MyShop:Inventory" in App Configuration: https://{account_name}.table.core.windows.net/{table_name}
+                                string tableContent = await ReadTableContentAsync(new Uri(setting.Value));
```

Comment:
It works but it seems like an odd use of table to me. This is more like blob usage. So my suggestion would either be to update to use blob storage, or update the reference to include table specific identifiers like partition and row key.

Reply:
On second thought. I think this is fine.